src/software/jetson_nano/services/power.cpp

```diff
@@ -5,25 +5,27 @@
 #include <cstdint>
 
 #include "proto/power_frame_msg.nanopb.h"
+#include "software/jetson_nano/services/network/power_service_exception.h"
 
 PowerService::PowerService()
 {
     if (!boost::filesystem::exists(DEVICE_SERIAL_PORT))
     {
-        LOG(FATAL) << "PLUG THE USB INTO THE JETSON NANO";
+        throw PowerServiceException("USB not plugged into the Jetson Nano");
     }
     this->uart = std::make_unique<BoostUartCommunication>(BAUD_RATE, DEVICE_SERIAL_PORT);
     this->read_thread = std::thread(boost::bind(&PowerService::continuousRead, this));
 }
 
 PowerService::~PowerService()
 {
+    is_running = true;
     read_thread.join();
 }
 
 void PowerService::continuousRead()
 {
-    for (;;)
+    while (!is_running)
```

Comment:
is_running logic seems flipped here. When is_running = true power service doesnt continuously read.

Reply:
Updated the code so that logic is clearer